File: reverseWhoisApi/password/nodejs/reverse_api_example.post.js
var https = require('https');

// Fill in your details
var username = 'your_whois_api_username';
var password = 'your_whois_api_password';

// Build the post string
var post_data ='{                       \
    "terms":[                           \
        {                               \
            "section":"Admin",          \
            "attribute":"name",         \
            "value":"Brett Branch",     \
            "exclude":"false"           \
        },                              \
        {                               \
            "section":"General",        \
            "attribute":"DomainName",   \
            "value":".com",             \
            "exclude":"false",          \
            "matchType":"EndsWith"      \
        }],                             \
    "recordsCounter":false,             \
    "outputFormat":"json",              \
    "username": "' + username + '",     \
    "password": "' + password +'",      \
    "rows":100                          \
}';

// set options fo request
var options = {
    hostname: 'www.whoisxmlapi.com',
    path: '/reverse-whois-api/search.php',
    port: 443,
    method: 'POST',
    headers: {
        'Content-Type': 'application/json',
        "Accept": "application/json",
        'Content-Length': post_data.length
    },
    json: true
};

// create request and get response
var req = https.request(options, function(res)  {
    var str = '';
    res.on('data', function(chunk) { str+=chunk; });
    res.on('end', function() { console.log(str); });

});

// handle errors
req.on('error', function(e) { console.error(e);});

// send request
req.write(post_data);
req.end();

File: assets/samples/reverse.whois/key.value/perl/reverse_api_example.post.pl
#!/usr/bin/perl

use strict;
use warnings;

use LWP::UserAgent;             # From CPAN

my $user_name = "Your whois api username";
my $password = "Your whois api password";
my $url = 'https://www.whoisxmlapi.com/reverse-whois-api/search.php';
my $content = '{"terms":[{"section":"Admin", "attribute":"organization",
    "value":"Wikimedia", "matchType":"anywhere", "exclude":"false"},
    {"section": "technical", "attribute": "city", "value": "San Francisco",
    "matchType":"anywhere", "exclude":"false"}], "recordsCounter":false,
    "outputFormat":"json", "username":"'.$user_name.'", 
    "password":"'.$password.'", "rows":100
}';

my $ua = LWP::UserAgent->new(ssl_opts => { verify_hostname => 0 });
print "\n" . $ua->post($url, Content => $content)->decoded_content . "\n";